x-pack/plugins/siem/public/components/open_timeline/index.tsx

```diff
@@ -261,4 +308,160 @@ export class StatefulOpenTimeline extends React.PureComponent<Props, State> {
+                  {}
+                )
+              : {},
+            pinnedEventsSaveObject: duplicate
+              ? {}
+              : timelineModel.pinnedEventsSaveObject != null
+              ? timelineModel.pinnedEventsSaveObject.reduce(
+                  (acc, pinnedEvent) => ({ ...acc, [pinnedEvent.pinnedEventId]: pinnedEvent }),
+                  {}
+                )
+              : {},
+            savedObjectId: duplicate ? null : timelineModel.savedObjectId,
+            version: duplicate ? null : timelineModel.version,
+            title: duplicate ? '' : timelineModel.title || '',
+          },
+        });
+        if (!duplicate) {
+          addNotes({
+            notes:
+              notes != null
+                ? notes.map<Note>(note => ({
+                    created: note.created != null ? new Date(note.created) : new Date(),
+                    id: note.noteId,
+                    lastEdit: note.updated != null ? new Date(note.updated) : new Date(),
+                    note: note.note || '',
+                    user: note.updatedBy || 'unknown',
+                    saveObjectId: note.noteId,
+                    version: note.version,
+                  }))
+                : [],
+          });
+        }
+      })
+      .finally(() => {
+        updateIsLoading({ id: 'timeline-1', isLoading: false });
+      });
+  };
+
+  private deleteTimelines: DeleteTimelines = (
+    timelineIds: string[],
+    variables?: AllTimelinesVariables
+  ) => {
+    if (timelineIds.includes(this.props.timeline.savedObjectId || '')) {
+      this.props.createNewTimeline({ id: 'timeline-1', columns: defaultHeaders, show: false });
+    }
+    this.props.apolloClient.mutate<
+      DeleteTimelineMutation.Mutation,
+      DeleteTimelineMutation.Variables
+    >({
+      mutation: deleteTimelineMutation,
+      fetchPolicy: 'no-cache',
+      variables: { id: timelineIds },
+      refetchQueries: [
+        {
+          query: allTimelinesQuery,
+          variables,
+        },
+      ],
+    });
+  };
 }
+
+const makeMapStateToProps = () => {
+  const getTimeline = timelineSelectors.getTimelineByIdSelector();
+  const mapStateToProps = (state: State) => {
+    const timeline = getTimeline(state, 'timeline-1');
+
+    return {
+      timeline,
+    };
+  };
+  return mapStateToProps;
+};
+
+export const StatefulOpenTimeline = connect(
+  makeMapStateToProps,
+  {
+    addTimeline: dispatchAddTimeline,
+    addNotes: dispatchAddNotes,
+    createNewTimeline: dispatchCreateNewTimeline,
+    setTimelineRangeDatePicker: dispatchSetTimelineRangeDatePicker,
+    updateIsLoading: dispatchUpdateIsLoading,
+  }
+)(StatefulOpenTimelineComponent);
+
+const omitTypename = (key: string, value: keyof TimelineModel) =>
+  key === '__typename' ? undefined : value;
+
+const omitTypenameInTimeline = (timeline: TimelineResult): TimelineResult =>
+  JSON.parse(JSON.stringify(timeline), omitTypename);
```

Comment:
For this stringify, I would add a try catch to guard against circular exceptions to a bit on the safer side for the stringify.